graveler/graveler.go

```diff
@@ -842,24 +842,14 @@ func newStagingToken(repositoryID RepositoryID, branchID BranchID) StagingToken
 	return StagingToken(v)
 }
 
-func (g *graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, branchID BranchID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
-	cancel, err := g.branchLocker.AquireMetadataUpdate(repositoryID, branchID)
-	if err != nil {
-		return "", fmt.Errorf("acquire metadata update: %w", err)
-	}
-	defer cancel()
+func (g *graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, parentCommitID CommitID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
 	repo, err := g.RefManager.GetRepository(ctx, repositoryID)
 	if err != nil {
 		return "", fmt.Errorf("get repository %s: %w", repositoryID, err)
 	}
-	branch, err := g.RefManager.GetBranch(ctx, repositoryID, branchID)
+	_, err = g.RefManager.GetCommit(ctx, repositoryID, parentCommitID)
```

Comment:
Think we should consider import to an empty repository - there will be no parent commit, or you should consider empty parent commit as valid.
Probably need to ignore the error ErrCommitNotFound and when filling the commit information, do not fill Parents with slice with empty string.